backend/utilities/rate_limiter.py

```
@@ -260,3 +262,31 @@ def check_test_login_rate_limit(request: Request) -> RateLimitResult:
+        retries: Number of retry attempts
+        delay: Initial delay between retries in seconds
+    """
+    def decorator(func: Callable):
+        @wraps(func)
+        async def wrapper(*args, **kwargs):
+            last_exception = None
+            
+            for attempt in range(retries + 1):
+                try:
+                    return await func(*args, **kwargs)
+                except Exception as e:
```

Ruff (0.14.0)

Do not catch blind exception: `Exception`

(BLE001)